src/xrpld/ledger/detail/View.cpp

```diff
@@ -1256,6 +1341,96 @@ accountSend(
     return terResult;
 }
 
+static TER
+rippleSendMPT(
+    ApplyView& view,
+    AccountID const& uSenderID,
+    AccountID const& uReceiverID,
+    STAmount const& saAmount,
+    STAmount& saActual,
+    beast::Journal j,
+    WaiveTransferFee waiveFee)
+{
+    assert(uSenderID != uReceiverID);
+
+    // Safe to get MPT since rippleSendMPT is only called by accountSendMPT
+    auto const issuer = saAmount.getIssuer();
+
+    if (uSenderID == issuer || uReceiverID == issuer || issuer == noAccount())
+    {
+        // if sender is issuer, check that the new OutstandingAmount will not
+        // exceed MaximumAmount
+        if (uSenderID == issuer)
+        {
+            auto const mptID =
+                keylet::mptIssuance(saAmount.get<MPTIssue>().getMptID());
+            auto const sle = view.peek(mptID);
+            if (!sle)
+                return tecMPT_ISSUANCE_NOT_FOUND;
+
+            if (sle->getFieldU64(sfOutstandingAmount) + saAmount.mpt().value() >
+                (*sle)[~sfMaximumAmount].value_or(maxMPTokenAmount))
```

Comment:
I don't think overflow is possible because we're only doing addition, but this feels a little hinky.
```suggestion
            auto const sendAmount = saAmount.mpt().value();
            auto const maximumAmount =
                sle->at(~sfMaximumAmount).value_or(maxMPTokenAmount);
            if (sendAmount > maximumAmount ||
                sle->getFieldU64(sfOutstandingAmount) >
                    maximumAmount - sendAmount)
```
If you don't like this change, could you add a comment pointing out that overflow is impossible, so future dummies like me don't get hung up on it?